syntax = "proto3";

package parse.v1;

import "common/v1/s3.proto";


message PDFCover {
  uint32 page_num = 1 ;
  repeated double bbox = 2 ;
}

message PDFSegment {
  enum SegmentType {
    TYPE_UNSPECIFIED = 0;
    TYPE_TEXT = 1;
    TYPE_IMAGE = 2;
    TYPE_FORMULA = 3; // 特指单行公式
    TYPE_TABLE = 4;
  }
  string chapter_name = 1 ; // 图片不存在 chapter name
  string id = 2 ;
  SegmentType type = 3 ;
  string content = 4 ; // 图片的这个部分为空，表格为解析版
  common.v1.S3File image = 5 ; // 图片格式数据
  repeated PDFCover covers = 6 ;
  repeated string related_segment_ids = 7 ;
  string reference_id = 8 ; // 图片和表格具有该字段
  string caption = 9 ; // 图片和表格具有该字段
}

message PDFMetadata {
  string title = 1 ;
}

message PDFParserInput {
  common.v1.S3File file = 1 ;
}

message PDFParserOutput {
  common.v1.S3File file = 1 ;
  PDFMetadata metadata = 2 ;
  repeated PDFSegment segments = 3 ;
}
